File: gccws-ui/src/app/main/module/super-admin/approve-control/service/approval-history.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {AppCrudRequestService} from "../../../../core/service/app-crud-request.service";
import {AppUtils} from "../../../../core/utils/app.utils";
import {AppConstants} from "../../../../core/constants/app.constants";
import {Observable} from "rxjs";
import {CommonResponseList, CommonResponseObject} from "../../../../core/model/common-response";
import {ApprovalHistory} from "../model/approval-history";

@Injectable({
    providedIn: 'root'
})
export class ApprovalHistoryService extends AppCrudRequestService<ApprovalHistory> {

    constructor(
      private http: HttpClient,
      private appUtils: AppUtils,
      private appConstants: AppConstants,
    ) {
        super(http, appUtils.getPrivateBaseURL() + appConstants.moduleSuperAdmin + 'approval-history');
    }


  getNotificationListByUserId(): Observable<CommonResponseList<ApprovalHistory>> {
    return this.httpClient.get<CommonResponseList<ApprovalHistory>>( this._BASE_URL + '/' + 'get-notification-list-by-user-id');
  }

  getApprovalPendingList(userId: number, fromDate: Date, toDate: Date, transactionTypeId: number): Observable<CommonResponseList<ApprovalHistory>> {
    return this.httpClient.get<CommonResponseList<ApprovalHistory>>( this._BASE_URL + '/' + 'get-pending-list-by-user-id' + '/' + userId+ '/' + fromDate + '/' + toDate + '/' + transactionTypeId);
  }

  getByApprovalHistoryId(approvalHistoryId: number,): Observable<CommonResponseList<ApprovalHistory>> {
    return this.httpClient.get<CommonResponseList<ApprovalHistory>>( this._BASE_URL + '/' + 'get-by-approval-history-id' + '/' + approvalHistoryId);
  }

  getLoadListData(userId: number, formDate: Date, toDate:Date): Observable<CommonResponseList<ApprovalHistory>> {
    return this.httpClient.get<CommonResponseList<ApprovalHistory>>( this._BASE_URL +'/' + userId + '/' + formDate + '/' + toDate  );
  }

}

File: gccws-server/sysadmin/src/main/java/com/gccws/sysadmin/approve/service/ApprovalHistoryService.java
package com.gccws.sysadmin.approve.service;

import com.gccws.base.service.BaseService;
import com.gccws.common.model.CommonPageableData;
import com.gccws.sysadmin.approve.dto.ApprovalHistoryDto;
import com.gccws.common.model.CommonDropdownModel;

import java.util.Date;
import java.util.List;

/**
 * @Author    Rima
 * @Since     February 23, 20223
 * @version   1.0.0
 */
public interface ApprovalHistoryService extends BaseService<ApprovalHistoryDto, CommonDropdownModel, CommonPageableData> {
    List<ApprovalHistoryDto> getNotificationListByNotifyUserId(Integer userId);
    List<ApprovalHistoryDto> getApprovalPendingList(Integer userId, Date fromDate, Date toDate, Integer transactionTypeId);
    List<ApprovalHistoryDto> getByApprovalHistoryId(Integer approvalHistoryId, Integer userId);

    List<ApprovalHistoryDto> getApprovalPendingListByDate(int userId, Date formDate, Date toDate );

}
